### PR TITLE
Goto map

### PR DESCRIPTION
**Pull request by Bruce-Connor**
This implements a new keymap, currently called `goto-map`, for moving
between different major-modes. This is inspired on the websites `g`
hotkey. You can `g m` to go between meta and main site, for instance

The problem is that I don't know which key to use for this keymap. We
can't use `g`, that already stands for `refresh` (which is pretty
widespread in Emacs).

I thought of `j`, for jump, but that's also taken.

### CODE REVIEW

**Comment by vermiculus** [comment]
Why not a capital `M` for 'Meta'?

**Comment by Bruce-Connor** [comment]
I like s. Can I use `s` and push `switch-site` to `s s`?

**Comment by vermiculus** [comment]
Sure thing :+1:

**Comment by Bruce-Connor** [comment]
Ok, I think I've finished adding everything I can think of.
Feel free to merge if you're good with the extra changes.

**Comment by Bruce-Connor** [comment]
Ok, I'm done adding everything I can think of here. Feel free to merge if you agree.

- Changed the file and keymap names to `switchto`. 
- Moved sx-search to `S`, sx-question-list-switch-site to `s s`, and made it so that the keybind only exists in buffers where `sx-question-list--site` is non-nil. A similar logic can also be used for other keys, if it makes sense.

File: sx-switchto.el [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+(mapc (lambda (x) (define-key sx-switchto-map (car x) (cadr x)))
+  '(
+    ;; These immitate the site's G hotkey.
+    ("a" sx-ask)
+    ("h" sx-tab-frontpage)
+    ("m" sx-tab-meta-or-main)
+    ;; This is `n' on the site.
+    ("u" sx-tab-unanswered)
+    ;; These are extra things we can do, because we're awesome.
```
Comment (vermiculus) [inline sx-switchto.el]: :+1:

File: sx-tab.el [hunk 2]
```diff
@@ -189,5 +189,62 @@ If SITE is nil, use `sx-default-site'."
...
+
+

+;;; Unanswered My-tags
+(sx-tab--define "Unanswered-my-tags"
+  (lambda (page)
+    (sx-question-get-questions
+     sx-question-list--site page nil 'unanswered/my-tags)))
+;;;###autoload
+(autoload 'sx-tab-unanswered
```
Comment (vermiculus) [inline sx-tab.el]: Was this a copy/paste error?

File: sx-tab.el [hunk 3]
```diff
@@ -189,5 +189,62 @@ If SITE is nil, use `sx-default-site'."
...
+;;; Inter-modes navigation
+(defun sx-tab-meta-or-main ()
+  "Switch to the meta version of a main site, or vice-versa.
+Inside a question, go to the frontpage of the site this question
+belongs to."
+  (interactive)
+  (if (and (derived-mode-p 'sx-question-list-mode)
+           sx-question-list--site)
+      (sx-question-list-switch-site
+       (if (string-match "\\`meta\\." sx-question-list--site)
```
Comment (vermiculus) [inline sx-tab.el]: Do we have to do regular expression matching here? I think matching the exact sequence would be faster.

**Comment by vermiculus** [comment]
Sorry, I saw your additional comment on the PR and assumed I'd read everything up to that point. I made a couple small comments, but I would draw your attention to the autoload issue. Some sort of copy/paste error, perhaps?

I restored the branch so we can push the changes again to merge.

File: sx-tab.el (hunk 2 above)
Comment (Bruce-Connor) [inline sx-tab.el]: Yes. I'll hotfix.

File: sx-tab.el (hunk 3 above)
Comment (Bruce-Connor) [inline sx-tab.el]: We could do a `(string= "meta." (substring sx-question-list--site 0 5))`, but that has to create a new string, so I'm not sure which's faster.

Comment (Bruce-Connor) [inline sx-tab.el]: What I do know is that `string-match` is the shortest option. Emacs doesn't have a “non-regexp” alternative to `string-match`.

**Comment by Bruce-Connor** [comment]
I pushed to this branch but nothing happened with this PR. I had to create a new PR.
Maybe I did something wrong. =/

**Comment by vermiculus** [comment]
It very well might have been deleting/restoring the branch – most likely not your fault :)

**Pull request merged**